extensions/amp-list/0.1/amp-list.js

```diff
@@ -387,15 +390,41 @@ export class AmpList extends AMP.BaseElement {
           AmpEvents.DOM_UPDATE, /* detail */ null, {bubbles: true});
       this.container_.dispatchEvent(event);
 
-      // Change height if needed.
-      this.measureElement(() => {
-        const scrollHeight = this.container_./*OK*/scrollHeight;
-        const height = this.element./*OK*/offsetHeight;
-        if (scrollHeight > height) {
-          this.attemptChangeHeight(scrollHeight).catch(() => {});
-        }
-      });
+      if (!isExperimentOn(this.win, 'amp-list-resize')) {
+        // Change height if needed.
+        this.measureElement(() => {
+          const scrollHeight = this.container_./*OK*/scrollHeight;
+          const height = this.element./*OK*/offsetHeight;
+          if (scrollHeight > height) {
+            this.attemptChangeHeight(scrollHeight).catch(() => {});
+          }
+        });
+      }
     }, this.container_);
+
+    if (isExperimentOn(this.getWin(), 'amp-list-resize')) {
+      this.mutateElement(() => {
+        this.changeToLayoutContainer_();
+      });
+    }
+  }
+
+  /**
+   * Converts the amp-list to de facto layout container.
+   * @private
+   */
+  changeToLayoutContainer_() {
+    this.element.setAttribute('layout', 'container');
+    const sizer = this.element.querySelector('i-amphtml-sizer');
+    if (sizer) {
+      this.element.removeChild(sizer);
+    }
+    this.container_.classList.remove('i-amphtml-fill-content');
+    this.container_.classList.remove('i-amphtml-replaced-content');
+    const overflowElement = this.element.querySelector('[overflow]');
```

Comment:
curious why this is needed. Technically after the resize `[overflow]` should become completely hidden. Whether it is left in the DOM or not should not impact anything.

Reply:
I'll investigate this. I think it was because in one of my test cases, the overflow element would not go away by itself, but I should clarify which one and why.